### ISSUE TITLE
[UI] Confirmation dialog when deleting

### ISSUE THREAD

**Issue by helfi92** [taskcluster/taskcluster#965]
Destructive actions (e.g., a delete operation) should have a confirmation step.
It’s an added click but will prevent accidental data loss frustration.

Forms that require a confirmation step dialog:
* [x] Role View (e.g., https://taskcluster-ui.herokuapp.com/auth/roles/hook-id%3Agarbage%2Fgecko-cinnabar-mirror)
* [x] Denylist (e.g., https://taskcluster-ui.herokuapp.com/notify/denylist/jan%40mozilla.com)
* [x] Clients View (e.g., https://taskcluster-ui.herokuapp.com/auth/clients/email%2Fbastien%40nextcairn.com%2Ftc-notify-test)
* [x] Secret View
    - You would first need to create a secret in https://taskcluster-ui.herokuapp.com/secrets before being able to view it and delete it.

Tip 1: In order to tackle this issue, I would recommend sending a PR for one of the forms to make sure the implementation is correct before adding that change to the other forms.

Tip 2: We can use the `DialogAction` component in `src/components/DialogAction` for the confirmation component.

**Comment by shubhamgupta2956** [comment]
@helfi92 I want to take this issue . Please assign it to me. Thanks.

**Comment by helfi92** [comment]
\o/ we are half way done :)

**Comment by shubhamgupta2956** [comment]
@helfi92 I tried opening denylist and adding secret but it is restricting me asking to add clientID. How can I add it?

**Comment by helfi92** [comment]
You will need to use the CLI to get a TC client and access token. The instructions can be found [here](https://github.com/taskcluster/taskcluster/tree/master/services/web-server#taskcluster-credentials). Let me know if that helps.

**Comment by shubhamgupta2956** [comment]
@helfi92 I am a bit confused on how to use taskcluster-cli. Like how to use it for adding those variables.

**Comment by helfi92** [comment]
taskcluster-cli is used to generate taskcluster credentials. Once you have the credentials, you click on the "sign in" button in the UI and then fill the clientID and access token with whathever taskcluster-cli creates for you. You can ignore the certificate input.

<img width="719" alt="Screen Shot 2019-07-19 at 8 48 28 PM" src="https://user-images.githubusercontent.com/3766511/61572061-9a158b00-aa66-11e9-8b0b-4e48ebd31dc6.png">

Does that help?

**Comment by shubhamgupta2956** [comment]
Sorry for not making the question clear. I just want to know about the commands or methods from which I can use taskcluster-cli to generate credentials. There are no instructions for that in the taskcluster-cli repo.

**Comment by helfi92** [comment]
Ah, the instructions are in the README of web-server. https://github.com/taskcluster/taskcluster/tree/master/services/web-server#taskcluster-credentials

Basically `./taskcluster signin --name taskcluster-web-server` would open a browser for you to login. Once you login via the browser, credentials will be outputted in the terminal.

**Comment by shubhamgupta2956** [comment]
I tried the command
```eval $(./taskcluster signin --name taskcluster-web-server)```
but it is not opening any browser. It just gives an error saying ```No such file or directory```

**Comment by helfi92** [comment]
You need to download the binary for your OS from the taskcluster-cli README

**Comment by shubhamgupta2956** [comment]
It is not available for linux.

**Comment by helfi92** [comment]
In https://github.com/taskcluster/taskcluster/tree/master/clients/client-shell#installation, there's linux-amd64. Does that work?

**Comment by shubhamgupta2956** [comment]
No, it is not working.

**Comment by helfi92** [comment]
Interesting. Any chance you are running the command outside the directory of where the binary is installed? I would expect a different error than `No such file or directory` if the binary is not for your OS.

**Comment by shubhamgupta2956** [comment]
No, I mean to say that the linux-amd64 file is not available in the site. It is giving github page 404 error.

**Comment by helfi92** [comment]
Oh I see. Does https://github.com/taskcluster/taskcluster/releases/tag/v15.0.0 work?

**Comment by shubhamgupta2956** [comment]
It is showing this error
![Screenshot from 2019-07-26 15-24-45](https://user-images.githubusercontent.com/43504292/61943657-8ebbe580-afb9-11e9-87a0-c7471bb2774f.png)

**Comment by helfi92** [comment]
@djmitche Any idea how @shubhamgupta2956 can get unblocked?

**Comment by shubhamgupta2956** [comment]
@helfi92 can I take another issue until I got unblocked?

**Comment by helfi92** [comment]
@djmitche Just informed me that the download link is not working and that we should be using https://github.com/taskcluster/taskcluster/releases/download/v15.0.0/taskcluster-linux-amd64 instead. See if that helps.

**Comment by helfi92** [comment]
Also, feel free to create a PR to fix the link in the README. I think there are extra quotes by mistake: https://github.com/taskcluster/taskcluster/tree/master/clients/client-shell#installation

**Comment by shubhamgupta2956** [comment]
It is working now but it still does not give me access token though it gives a clientId to me. Also when I visited the denylist page, it says "Bad mac". How can I solve this?

**Comment by helfi92** [comment]
Bad mac means you are using bad credentials (probably the wrong access token for the client ID). What command did you run on your terminal?

**Comment by shubhamgupta2956** [comment]
export TASKCLUSTER_ROOT_URL="https://taskcluster.net" eval $(./taskcluster-linux-amd64 signin --name taskcluster-web-server)

**Comment by helfi92** [comment]
> It is working now but it still does not give me access token though it gives a clientId to me

This is weird. Where is it giving you the client ID. In the terminal or browser? Can you provide a screenshot of where it's giving it to you, please?

**Comment by shubhamgupta2956** [comment]
It gives the access token this time. I used mozilla oAuth last time and it does not provides any token. I don't know how it worked this time. But I now got the scope problem and when I tried to add it, it says that I am not authorized.
![Screenshot from 2019-08-01 08-38-03](https://user-images.githubusercontent.com/43504292/62262883-e7ff9b00-b437-11e9-8a83-371465a3bb35.png)

**Comment by helfi92** [comment]
Try again. I just gave you that missing scope.

**Comment by shubhamgupta2956** [comment]
I got disabled now.
![Screenshot from 2019-08-01 21-27-19](https://user-images.githubusercontent.com/43504292/62308582-35652200-b4a3-11e9-87f3-796a3363a1ea.png)

**Comment by helfi92** [comment]
I'm not sure why. @djmitche Would you happen to know by any chance why @shubhamgupta2956 client got disabled?

**Comment by shubhamgupta2956** [comment]
Can there be any problem with the clientId?
My clientId is `mozilla-auth0/oauth2|firefoxaccounts|6de766304ad44497b89bbe624c0da144|shubham.gupta2956@gmail.com/`

**Comment by helfi92** [comment]
If you navigate to https://taskcluster-ui.herokuapp.com/auth/clients?search=mozilla-auth0%2Foauth2%7Cfirefoxaccounts%7C6de766304ad44497b89bbe624c0da144 you will see that you've created many client IDs. The client ID with the required scopes is the one in https://github.com/taskcluster/taskcluster/issues/965#issuecomment-517100620 since that's the only I gave permission. All the others will complain about missing scopes.

**Comment by shubhamgupta2956** [comment]
The client ID is not in the link you have provided. Please remove the scope from the previous client ID and add it in https://github.com/taskcluster/taskcluster/issues/965#issuecomment-517387079. Sorry for inconvenience

**Comment by djmitche** [comment]
> I'm not sure why. @djmitche Would you happen to know by any chance why @shubhamgupta2956 client got disabled?

The client scanner probably did that.

**Comment by helfi92** [comment]
Done. Added the scopes to https://taskcluster-ui.herokuapp.com/auth/clients/mozilla-auth0%2Foauth2|firefoxaccounts|6de766304ad44497b89bbe624c0da144|shubham.gupta2956%40gmail.com%2F.

**Comment by shubhamgupta2956** [comment]
I am still disabled...
![Screenshot from 2019-08-03 20-28-17](https://user-images.githubusercontent.com/43504292/62413504-548fbb00-b62d-11e9-8aef-d3aca64e514f.png)

**Comment by shubhamgupta2956** [comment]
@helfi92 can I take any other issue until this got fixed?

**Comment by helfi92** [comment]
@shubhamgupta2956 Are you able to create then view a secret owned by you?

1. Login
2. Create a secret
3. View the secret you just created

**Comment by shubhamgupta2956** [comment]
No, I can't add secret. It is still showing the user is disabled.

**Comment by helfi92** [comment]
Do you mind trying again. I just created a role with the scopes: https://tools.taskcluster.net/auth/roles/login-identity%3Amozilla-auth0%2Foauth2|firefoxaccounts|6de766304ad44497b89bbe624c0da144|shubham.gupta2956%40gmail.com

If that doesn't work, what clientId is it saying that it's disabled?

**Comment by shubhamgupta2956** [comment]
It is still not working. Can it be due to roleID instead of clientID?

**Comment by helfi92** [comment]
What clientId is it saying that it's disabled?

**Comment by helfi92** [comment]
Actually. I just enabled your clientId. Not sure why it got disabled. It should work now.

**Comment by shubhamgupta2956** [comment]
It is now saying
`Client with clientId 'mozilla-auth0/oauth2|firefoxaccounts|6de766304ad44497b89bbe624c0da144|shubham.gupta2956@gmail.com' not found`

**Comment by helfi92** [comment]
Right it doesn't exist. The one that exists ends with a forward slash at the end of the name.

**Comment by shubhamgupta2956** [comment]
I think I had misplaced the access key for this clientID. How can I get it again?

**Comment by helfi92** [comment]
You'll need to run the command in https://github.com/taskcluster/taskcluster/issues/965#issuecomment-514379895, click on "Create a new clientId", then edit the client ID name accordingly.

![Screen Shot 2019-08-09 at 5 55 51 PM](https://user-images.githubusercontent.com/3766511/62811230-f8b2af80-bace-11e9-98ec-90e2ebd24422.png)

**Comment by helfi92** [comment]
Actually, if you probably just need to run the command in the previous comment then login in the UI with good credentials. The role that I setup should allow that client to work.

**Comment by shubhamgupta2956** [comment]
But how can I login if I don't have access token? Also I can't access that page until I am not logged in.

**Comment by helfi92** [comment]
> But how can I login if I don't have access token?

**1)** Run `./taskcluster signin --name taskcluster-web-server` in the terminal. Wait for a new browser tab to open.

**2a)** If you are already logged in to tools.taskcluster.net, you will see:

![Screen Shot 2019-08-10 at 9 29 09 AM](https://user-images.githubusercontent.com/3766511/62822436-82a35c80-bb51-11e9-90c3-c37d7f8c62fa.png)

Click on "Create a new ClientID" and then create a client ID.

**2b)** If you are not logged in, you will see:

![Screen Shot 2019-08-10 at 9 29 52 AM](https://user-images.githubusercontent.com/3766511/62822446-a23a8500-bb51-11e9-813b-685e80f9942f.png)

Login via Auth0.
 
**3)** Once you log in, you should see a page with the following content:

![Screen Shot 2019-08-10 at 9 34 01 AM](https://user-images.githubusercontent.com/3766511/62822464-0d845700-bb52-11e9-94c7-febdad89e317.png)

**4)** Look at your terminal, it should have outputted `TASKCLUSTER_CLIENT_ID` and `TASKCLUSTER_ACCESS_TOKEN`.

**5** Navigate to the new UI http://localhost:5080/ and login with these new credentials.

Let me know if this helps.

**Comment by shubhamgupta2956** [comment]
It worked and now I can access both the pages. I will try to finish this task ASAP.

**Comment by shubhamgupta2956** [comment]
I am now getting this
![Screenshot from 2019-08-11 21-45-46](https://user-images.githubusercontent.com/43504292/62836578-5fc49580-bc82-11e9-858e-dd83b0f6cd9d.png)
Do I have to generate a separate access token every time?

**Comment by helfi92** [comment]
I extended the expiration of the client ID. It should work now :)

**Comment by shubhamgupta2956** [comment]
@helfi92 I am getting this. I can't detect the problem.
![Screenshot from 2019-08-13 03-53-39](https://user-images.githubusercontent.com/43504292/62903081-32a9dd00-bd7f-11e9-9263-e136cc7acfd8.png)

**Comment by arku** [comment]
@shubhamgupta2956 It would be helpful if you could push your changes and share a link to your branch. If I have to guess, it looks like you're trying to render an object. Something like

```jsx
<div>
  {something}
<div>
```

where `something` is an object. 

Does that help?

**Comment by helfi92** [comment]
One more view to go \o/

**Comment by helfi92** [comment]
Amazing work @shubhamgupta2956! We are very grateful for your help 🥇

**Comment by shubhamgupta2956** [comment]
Thanks :slightly_smiling_face:  . Can you please suggest me which issue should I pick next.

**Comment by helfi92** [comment]
Hey @Shubhamchinda, always great to hear from you :) 2 things I think you can help me with:

1. We have a new view that's missing a confirmation step when deleting http://localhost:5080/worker-manager/test-2%2Ftest. Adding a confirmation step similar to the other views you worked on would be really awesome.

2. The save button in the light theme looks weird http://localhost:5080/worker-manager/test-2%2Ftest

![Screen Shot 2019-08-30 at 9 10 31 AM](https://user-images.githubusercontent.com/3766511/64023332-0ecbf280-cb06-11e9-926a-042e91b767e4.png)

It should look similar to how the other views do it:

![Screen Shot 2019-08-30 at 9 11 23 AM](https://user-images.githubusercontent.com/3766511/64023381-2d31ee00-cb06-11e9-8859-cdc5c70a889e.png)

**Comment by helfi92** [comment]
Filed https://github.com/taskcluster/taskcluster/issues/1262 :)

**Comment by Shubhamchinda** [comment]
> Hey @Shubhamchinda, always great to hear from you :) 2 things I think you can help me with:
> 
> 1. We have a new view that's missing a confirmation step when deleting http://localhost:5080/worker-manager/test-2%2Ftest. Adding a confirmation step similar to the other views you worked on would be really awesome.
> 2. The save button in the light theme looks weird http://localhost:5080/worker-manager/test-2%2Ftest
> 
> ![Screen Shot 2019-08-30 at 9 10 31 AM](https://user-images.githubusercontent.com/3766511/64023332-0ecbf280-cb06-11e9-926a-042e91b767e4.png)
> 
> It should look similar to how the other views do it:
> 
> ![Screen Shot 2019-08-30 at 9 11 23 AM](https://user-images.githubusercontent.com/3766511/64023381-2d31ee00-cb06-11e9-8859-cdc5c70a889e.png)

Hey, @helfi92 , I'd love to work on it! 
Thanks a lot :)

**Comment by Shubhamchinda** [comment]
> Hey @Shubhamchinda, always great to hear from you :) 2 things I think you can help me with:
I think you tagged me by mistake. 
Let me know if you got anything else for me. :)

**Comment by helfi92** [comment]
😅 hehe sorry for that. I will definitely help you find an issue to work on :)
